tests/cnf/core/network/metallb/tests/frrk8-tests.go

```diff
@@ -370,12 +377,15 @@ var _ = Describe("FRR", Ordered, Label(tsparams.LabelFRRTestCases), ContinueOnFa
 				By("Verify node state updates on worker node 1")
 				Eventually(func() string {
 					// Get the routes
-					frrNodeState, err := metallb.ListFrrNodeState(APIClient)
+					frrNodeState, err := metallb.ListFrrNodeState(APIClient, client.ListOptions{
+						FieldSelector: fields.SelectorFromSet(fields.Set{"metadata.name": "worker-1"})})
 					Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP routes")
-					fmt.Println("frrNodeState[0].Objects.Status.RunningConfig",
-						frrNodeState[0].Object.Status.RunningConfig)
 
-					return frrNodeState[0].Object.Status.RunningConfig
+					frrWorker1Status := findNodeByName(frrNodeState, cnfWorkerNodeList[1].Definition.Name)
+					Expect(frrWorker1Status).ToNot(BeNil(), fmt.Sprintf("Node with name %s not found",
+						cnfWorkerNodeList[1].Definition.Name))
+
+					return frrWorker1Status.Object.Status.RunningConfig
```

Comment:
`return frrNodeState[0].Object.Status.RunningConfig`

Back to original

Reply:
changed